pkg/ebpftracer/tracer.go

```diff
@@ -545,14 +607,56 @@ func (t *Tracer) getPolicy(eventID events.ID, cgroupID uint64) *cgroupEventPolic
 	return nil
 }
 
-func (t *Tracer) removeCgroup(cgroupID uint64) {
-	t.policyMu.Lock()
-	delete(t.cgroupEventPolicy, cgroupID)
-	t.policyMu.Unlock()
+func (t *Tracer) cgroupCleanupLoop(ctx context.Context) error {
+	cleanupTimer := time.NewTicker(1 * time.Minute)
+	for {
+		select {
+		case <-ctx.Done():
+			cleanupTimer.Stop()
+			return ctx.Err()
+		case <-cleanupTimer.C:
+		}
+
+		now := time.Now()
+		t.cgroupCleanupMu.Lock()
+		toCleanup := t.requestedCgroupCleanups
```

Comment:
Took me some time to understand selecting by slice indexes. Could be as simple as

```go
now := time.Now()
toCleanup := lo.Filter(t.requestedCgroupCleanups, func(cg ..) bool {
   return cg.cleanupAt.Before(now)
})
```

Reply:
The difference is, that the loop just exits after finding the first cgroup to cleanup, as the slice will always be sorted. 

I will still use the `lo` solution though, as in the end this code runs every other minute, so it is more than efficient enough.

Reply:
Why not cleanup all cgroups in single go? If there are many temp containers create/delete and we remove only one cgroup in one minute we can potentially start lagging.

Reply:
One minute might be too long for this, you are right. The problem when cleaning up in one go is, that we still have two distinct loops. There is always the possibility of a timing issue that is causing containers/cgroups to be reloaded yielding a memory leak. 

Maybe it makes sense to store the cleanup queue as buckets, since then we would not need to figure where to split. The only downside, this would make the logic more complicated.